src/Juvix/Compiler/Core/Translation/FromInternal.hs

```diff
@@ -633,6 +624,12 @@ goExpression' = \case
       varsNum <- asks (^. indexTableVarsNum)
       return (mkVar (setInfoLocation (n ^. nameLoc) (Info.singleton (NameInfo (n ^. nameText)))) (varsNum - k - 1))
     Internal.IdenFunction n -> do
+      funInfoBuiltin <- getFunctionBuiltinInfo n
+      case funInfoBuiltin of
+        Just Internal.BuiltinBoolIf -> error "if must be called with 3 arguments"
```

Comment:
These should not be developer errors. We should throw a `JuvixError` before or at typechecking.
If you prefer it, we can address it in a separate pr

Reply:
Yes, I was intending to make a separate issue for this. I think these should be checked earlier - during type checking or arity checking. I did it this way because in the internal-to-core translation there was already a check like this for `if`, etc. at different places.